src/derive/curve.rs

```diff
@@ -49,12 +49,19 @@ macro_rules! endo {
     };
```

Comment:
I think the specification of the encoding must be copied here as a rust doc comment.  Just copying it would be fine.  This way we keep the explanation of how the serialization and deserialization works next to the code and can be read by anyone using the code.

On the other hand I have suggestions about improving the specification format, I think it requires explicit mention of:
- The fact that coordination encoding doesn't specify the endianness
- Every time a "bit nth" is mentioned, say "the nth most significant bit" or the nth msb" or something like that, to avoid confusion
- Indicate that all tables have the most significant bits on the left, and less significant bits on the right
  - This should clarify the comment https://github.com/privacy-scaling-explorations/halo2curves/pull/141#pullrequestreview-1908366172 because currently code for uncompressed format flips x/y compared to the specs.